cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -387,27 +391,18 @@ public final ProgramController run(final Program program, ProgramOptions oldOpti
   }
 
   /**
-   * For programs initiated by a tethered peer, load additional resources that were saved locally based by runId
+   * For programs initiated by a tethered peer, load additional resources that were saved based by runId
    */
-  private void loadLocalResources(Program program, CConfiguration cConf, String runId)
-    throws URISyntaxException, IOException {
-    File tmpDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
-                           cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
-    for (File file : DirUtils.listFiles(new File(tmpDir, runId))) {
-      if (file.getName().equals(LOGBACK_FILE_NAME)) {
-        // replace existing logback.xml file
-        URI logbackURI = getLogBackURI(program);
-        if (logbackURI != null) {
-          try (InputStream input = logbackURI.toURL().openStream()) {
-            Files.copy(input, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
-          }
-        }
-      } else if (file.getName().equals(TETHER_CONF_FILE_NAME)) {
-        // add additional cConf entries
-        cConf.addResource(file.toURI().toString());
-        cConf.reloadConfiguration();
-      }
+  private void loadAdditionalResources(Arguments args, CConfiguration cConf, File tempDir) throws IOException {
+    Location location = locationFactory.create(args.getOption(ProgramOptionConstants.PROGRAM_RESOURCE_URI));
+    // add additional cConf entries
+    File tetherCConfFile = File.createTempFile("cdap-tether", ".xml", tempDir);
+    try (InputStream is = location.append(TETHER_CONF_FILE_NAME).getInputStream()) {
+      Files.copy(is, tetherCConfFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+      LOG.debug("Copied {} to {}", TETHER_CONF_FILE_NAME, tetherCConfFile);
     }
+    cConf.addResource(tetherCConfFile.toURI().toURL());
+    cConf.reloadConfiguration();
```

Comment:
Do we need to call cConf.reloadConfiguration()? Looks like it's called as part of addResource() [here](https://github.com/cdapio/cdap/blob/433f41d8e682ba5d4592b6ec0d139c64970e3c40/cdap-common/src/main/java/io/cdap/cdap/common/conf/Configuration.java#L460)

Reply:
looks like the call is unneeded. removed